client/cli_crypto_service.go

```diff
@@ -95,9 +120,8 @@ func (ccs *CryptoService) Sign(keyIDs []string, payload []byte) ([]data.Signatur
 	return signatures, nil
 }
 
-func sign(privKey *data.PrivateKey, hash crypto.Hash, hashed []byte) ([]byte, error) {
-	// TODO(diogo): Implement support for ECDSA.
-	if strings.ToLower(privKey.Cipher()) != "rsa" {
+func rsaSign(privKey *data.PrivateKey, hash crypto.Hash, hashed []byte) ([]byte, error) {
+	if strings.ToUpper(privKey.Cipher()) != "RSA" {
```

Comment:
+1 for the string constants.

Having Cipher do the case conversion seems like it would be forcing a convention on everyone. Whatever somebody sets as the type when creating a key should be what they get back when calling Cipher.